src/kbemu.c

```diff
@@ -37,26 +39,13 @@ KeySym kbemu_keysym(const char *str)
 
 void send_kbevent(KeySym key, int press)
 {
-	XEvent xevent;
-	Window win;
-	int rev_state;
-
 	if(!dpy) return;
 
+	Window win;
+	int rev_state;
 	XGetInputFocus(dpy, &win, &rev_state);
 
-	xevent.type = press ? KeyPress : KeyRelease;
-	xevent.xkey.display = dpy;
-	xevent.xkey.root = DefaultRootWindow(dpy);
-	xevent.xkey.window = win;
-	xevent.xkey.subwindow = None;
-	xevent.xkey.keycode = XKeysymToKeycode(dpy, key);
-	xevent.xkey.state = 0;
-	xevent.xkey.time = CurrentTime;
-	xevent.xkey.x = xevent.xkey.y = 1;
-	xevent.xkey.x_root = xevent.xkey.y_root = 1;
-
-	XSendEvent(dpy, win, True, press ? KeyPressMask : KeyReleaseMask, &xevent);
+	XTestFakeKeyEvent(dpy, XKeysymToKeycode(dpy, key), press, 0);
```

Comment:
You can't rely on Xtest being always available. The existing XSendEvent code needs to be retained as a fallback, and the availability of Xtest needs to be queried at runtime during initialization.

Reply:
Thanks for the quick feedback! Is this for the case that the XTest extension is available at compile time but not at runtime? Do you know how I can then test for the availability at runtime?

Reply:
Yes, exactly. You can see if it's available by calling the `XTestQueryExtension` function.

Reply:
Did it. Can you please check the changes? Thanks!